pkg/deploy/skr_webhook_resources.go

```diff
@@ -0,0 +1,341 @@
+		webhook := registrationV1.ValidatingWebhook{
+			Name:                    fmt.Sprintf("%s.operator.kyma-project.io", moduleName),
+			ObjectSelector:          &metav1.LabelSelector{MatchLabels: watchableCfg.Labels},
+			AdmissionReviewVersions: []string{version},
+			ClientConfig: registrationV1.WebhookClientConfig{
+				CABundle: caCert,
+				Service: &registrationV1.ServiceReference{
+					Name:      svcObjKey.Name,
+					Namespace: svcObjKey.Namespace,
+					Path:      &svcPath,
+				},
+			},
+			Rules: []registrationV1.RuleWithOperations{
+				{
+					Rule: registrationV1.Rule{
+						APIGroups:   []string{v1alpha1.GroupVersion.Group},
+						APIVersions: []string{"*"},
+						Resources:   []string{watchableResources},
+					},
+					Operations: []registrationV1.OperationType{
+						"CREATE", "UPDATE", "DELETE",
+					},
+				},
+			},
+			SideEffects:    &sideEffects,
+			TimeoutSeconds: timeout,
+			FailurePolicy:  &failurePolicy,
+		}
+		webhooks = append(webhooks, webhook)
+	}
+	return &registrationV1.ValidatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ValidatingWebhookConfiguration",
+			APIVersion: registrationV1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      webhookObjKey.Name,
+			Namespace: webhookObjKey.Namespace,
+		},
+		Webhooks: webhooks,
+	}
+}
+
+func configureClusterRoleBinding(cfg *unstructuredResourcesConfig, binding *unstructured.Unstructured,
+) (*unstructured.Unstructured, error) {
+	subjects, found, err := unstructured.NestedSlice(binding.UnstructuredContent(), bindingSubjectsKey)
+	if !found {
+		return nil, ErrCouldNotFindSubjectsField
+	}
+	if err != nil {
+		return nil, err
+	}
+	if len(subjects) == 0 {
+		return nil, ErrExpectedSubjectsNotToBeEmpty
+	}
+	serviceAccountSubj, ok := subjects[0].(map[string]interface{})
+	if !ok {
+		return nil, ErrFailedToConvertSubjectToMap
+	}
+	if err := unstructured.SetNestedField(serviceAccountSubj, cfg.remoteNs, "namespace"); err != nil {
+		return nil, err
+	}
+	subjects[0] = serviceAccountSubj
+	if err := unstructured.SetNestedSlice(binding.Object, subjects, bindingSubjectsKey); err != nil {
+		return nil, err
+	}
+	return binding, nil
+}
+
+func configureConfigMap(cfg *unstructuredResourcesConfig, configMap *unstructured.Unstructured,
+) (*unstructured.Unstructured, error) {
+	configMapData := map[string]string{
+		"contractVersion":  cfg.contractVersion,
+		"kcpAddr":          cfg.kcpAddress,
+		"tlsWebhookServer": cfg.tlsWebhookServer,
+		"tlsCallback":      cfg.tlsCallback,
+	}
+	if err := unstructured.SetNestedStringMap(configMap.Object, configMapData, stringMapDataKey); err != nil {
+		return nil, err
+	}
+	return configMap, nil
+}
+
+func configureDeployment(cfg *unstructuredResourcesConfig, deployment *unstructured.Unstructured,
+) (*unstructured.Unstructured, error) {
+	err := unstructured.SetNestedField(deployment.Object, cfg.secretResVer, "spec", "template", "metadata",
```

Comment:
why do you not use the native APIs like v1.Deployment?

Reply:
Can we unmarshal the yaml to native API structs instead of Unstructured?

Reply:
yes this is possible. you can also convert from unstructured to the typed one with runtime.DefaultUnstructuredConverter if you prefer this.